pdns/speedtest.cc

```diff
@@ -505,18 +484,9 @@ static bool magicAddrMatch(const QType& query, const QType& answer)
 }
 
 
-bool moreSpecificThan(const string& a, const string &b)
+bool moreSpecificThan(const DNSName& a, const DNSName &b)
 {
-  static string dot(".");
-  int counta=(a!=dot), countb=(b!=dot);
-  
-  for(string::size_type n=0;n<a.size();++n)
-    if(a[n]=='.')
-      counta++;
-  for(string::size_type n=0;n<b.size();++n)
-    if(b[n]=='.')
-      countb++;
-  return counta>countb;
+  return (a.isPartOf(b) && a.countLabels() > b.countLabels());
```

Comment:
The old code did not do the equivalent of `isPartOf`, is it okay that you changed this?